lib/openxml/pptx/parts/slide_layout.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literals: true
+require "openxml/elements/notes_size"
+require "openxml/elements/slide_size"
+
+module OpenXml
+  module Pptx
+    module Parts
+      class SlideLayout < OpenXml::Part
+        attr_accessor :relationships
+        private :relationships=
+
+        def self.defualt_relationships
+          @default_relationships ||= []
+        end
+
+        def self.relationship(type, target)
+          self.defualt_relationships << [type, target]
+        end
+        relationship("http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideMaster",
+                     "../slideMasters/slideMasterBasic.xml")
+
+        def initialize
+          self.relationships = OpenXml::Parts::Rels.new
+
+          self.class.defualt_relationships.each do |type, target|
+            add_relationship type, target
+          end
+        end
+
+        def add_relationship(type, target)
+          relationships.add_relationship(type, target)
+        end
+
+        def add_to(ancestors)
+          parent, *rest = ancestors
+          parent.add_part rest, "slideLayouts/slideLayoutBasic.xml", self
```

Comment:
I've been debating about that. Generating makes things harder to debug and a little less human readable. Assigning names all over might get daunting for the user of the system. I'm not sure that I care. We could reserver the defaults with names, and then require a name if someone wants to make a new layout, master, theme?